src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryOperators.cs

```diff
@@ -809,6 +839,26 @@ private static FormulaValue TimeDifference(IRContext irContext, FormulaValue[] a
 
             var result = arg0.Value.Subtract(arg1.Value);
             return new NumberValue(irContext, result.TotalDays);
+        }
+
+        private static FormulaValue DateDifferenceDecimal(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
```

Comment:
>DateDifferenceDecimal [](http://example.com/codeflow?start=36&length=21)

You can merge those two new methods with the ones above, and check the irContext's return type (irContext.ResultType._type.Kind) to determine whether to create a NumberValue or a DecimalValue

Reply:
Done.